### ISSUE TITLE
Partially but incorrectly addressed SC2002 should raise a new issue

### ISSUE THREAD

**Issue by jstangroome** [koalaman/shellcheck#794]
Partly related to SC2002. Reproducible at http://www.shellcheck.net/.

Start with:
```sh
#!/bin/sh
cat file | grep hello
```
shellcheck correctly reports SC2002.

Fix part of the issue by replacing `cat` with `<` but forgetting to remove the `|`:
```sh
#!/bin/sh
<file | grep hello
```
shellcheck now reports `No issues detected!` but this script won't work as expected.

Maybe there is a scenario where this script is useful but I don't know the shell well-enough to be sure. It may be useful for shellcheck to report when an input redirect is not associated with any command before a pipe.

**Comment by koalaman** [comment]
Good call. A warning has been added in f594f01.
